### ISSUE TITLE
Add an `activation` argument for all classifiers

### ISSUE THREAD

**Issue by mattdangerw** [keras-team/keras-nlp#954]
For general usability and consistency with keras-cv, we would like to add an `activation` argument to all classifiers.

```python
model = BertClassifier.from_preset(
    "bert_base_en_uncased",
    num_classes=2,
    activation="softmax",
)
model.compile(...)
model.fit(...)
model.predict(["What an amazing movie!", "A total waste of my time."])
>>> np.array([[0.004173, 0.9956  ], [0.997   , 0.0028  ]])
```

One way this could work:
- `BertClassifier` activation defaults to `None` (logits).
- If activation is softmax, we default compile with `loss=keras.losses.SparseCategoricalCrossentropy(from_logits=False)`.
- If activation is `None`, we default compile with `loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True)`.
- If activation is something else, we do not default compile.
